### ISSUE TITLE
Can't build run rust-tensorflow

### ISSUE THREAD

**Issue by tyoc213** [Microsoft/WSL#1467]
Please use the following bug reporting template to help produce actionable and reproducible issues. Please try to ensure that the reproduction is minimal so that the team can go through more bugs!

* Can't build tensor flow or build/use rust-tensorflow

* Some output after `cargo build` after include the crate as test (yeah I have bazel installed and things like that)

```
$ cargo build
   Compiling tensorflow-sys v0.5.0
                                                                                      error: failed to run custom build command for `tensorflow-sys v0.5.0`
process didn't exit successfully: `/mnt/c/Users/tyoc/try2/lala/target/debug/build/tensorflow-sys-f5d6b3977d411f10/build-script-build` (exit code: 101)
--- stdout
libtensorflow-sys/build.rs:31: output = "/mnt/c/Users/tyoc/try2/lala/target/debug/build/tensorflow-sys-f5d6b3977d411f10/out"
libtensorflow-sys/build.rs:33: source = "/home/tyoc213/.cargo/registry/src/github.com-1ecc6299db9ec823/tensorflow-sys-0.5.0/target/source-0.10.0"
libtensorflow-sys/build.rs:35: lib_dir = "/mnt/c/Users/tyoc/try2/lala/target/debug/build/tensorflow-sys-f5d6b3977d411f10/out/lib-0.10.0"
libtensorflow-sys/build.rs:37: Directory "/mnt/c/Users/tyoc/try2/lala/target/debug/build/tensorflow-sys-f5d6b3977d411f10/out/lib-0.10.0" already exists
libtensorflow-sys/build.rs:43: library_path = "/mnt/c/Users/tyoc/try2/lala/target/debug/build/tensorflow-sys-f5d6b3977d411f10/out/lib-0.10.0/libtensorflow_c.so"
libtensorflow-sys/build.rs:48: target_path = "tensorflow/libtensorflow_c.so"
libtensorflow-sys/build.rs:77: Checking build file "/home/tyoc213/.cargo/registry/src/github.com-1ecc6299db9ec823/tensorflow-sys-0.5.0/target/source-0.10.0/tensorflow/BUILD"
libtensorflow-sys/build.rs:82: Build file "/home/tyoc213/.cargo/registry/src/github.com-1ecc6299db9ec823/tensorflow-sys-0.5.0/target/source-0.10.0/tensorflow/BUILD" already patched
libtensorflow-sys/build.rs:102: Executing "./configure"
No Google Cloud Platform support will be enabled for TensorFlow
No GPU support will be enabled for TensorFlow
Configuration finished
libtensorflow-sys/build.rs:106: Command "./configure" finished successfully
libtensorflow-sys/build.rs:102: Executing "bazel" "build" "--jobs=8" "--compilation_mode=opt" "tensorflow:libtensorflow_c.so"

--- stderr
........................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................
Error: couldn't connect to server at '/home/tyoc213/.cache/bazel/_bazel_tyoc213/03736e16cd598f65b37ea03f9c7e8e67/server/server.socket' after 60 seconds.
thread 'main' panicked at 'failed to execute "bazel" "build" "--jobs=8" "--compilation_mode=opt" "tensorflow:libtensorflow_c.so"', /home/tyoc213/.cargo/registry/src/github.com-1ecc6299db9ec823/tensorflow-sys-0.5.0/build.rs:104
note: Run with `RUST_BACKTRACE=1` for a backtrace.
```

* CurrentBuildNumber: 14393

* well, I cant provide all the commands taht I have run but the steps for build tensorflow would be enought... some like https://bazel.build/versions/master/docs/install.html#ubuntu from https://github.com/tensorflow/rust#automatically-building-tensorflow

```
$ curl https://sh.rustup.rs -sSf | sh
$ sudo add-apt-repository ppa:webupd8team/java
$ sudo apt-get update
$ sudo apt-get install oracle-java8-installer
# add the repos for bazel and update apt
$ echo "deb [arch=amd64] http://storage.googleapis.com/bazel-apt stable jdk1.8" | sudo tee /etc/apt/sources.list.d/bazel.list
$ curl https://bazel.build/bazel-release.pub.gpg | sudo apt-key add -
$ sudo apt-get update && sudo apt-get install bazel
```

Then make

```
 source $HOME/.cargo/env
```

because it seem to not execute it each time I run it with WIndows+R

```
$ cargo new test
```

Inside edit `Cargo.toml` and add the dependency

```
[dependencies]
tensorflow = "0.1"
```

And then inside the root of that test do

```
$ cargo build
```

Will provide strace later, I have an open issue there also for this https://github.com/tensorflow/rust/issues/36

By the way, I also did try building tensorflow as https://github.com/tensorflow/rust#manually-building-tensorflow but get the same result (about babel IIRC).

**Comment by aseering** [comment]
Hi @tyoc213 -- it looks like Bazel requires Java.  Linux Java is known to not work well on stable Windows.  It works much better in the latest [Insider build](https://insider.windows.com).  Would you be willing to try a more recent build?

**Comment by tyoc213** [comment]
Well, windows10upgrade9252.exe that is the one you can obtain in that page says that I already have the latest windows.... so I guess you mean I need to download an iso from here burn it and run it?? https://www.microsoft.com/en-us/software-download/windowsinsiderpreviewiso ???

**Comment by aseering** [comment]
@tyoc213 -- you can sign up for the "Fast" or "Slow" ring through the advanced settings in the "Check for Updates" app in Windows.  If you do, you'll start getting new builds as part of the normal Windows Update process.  (It can take a few days for the updates to start showing up.)

You can also download the ISO, if you don't want to wait.  Note that the ISO is for the "slow" ring -- it's a much more recent build, but it's not the very latest build.  The only way to get the very latest build is through the "Check for Updates" method.

Just so you know -- Insider builds are pre-release builds of all of Windows.  They have new features; also new bugs :-)

**Comment by tyoc213** [comment]
How I know Im updated to the latest? so that I can try to build it again?

**Comment by aseering** [comment]
The current latest build for PC is 14971.  You said that you were running 14393 above; you've successfully upgraded if that number says 14971.

New builds are announced here, if you're reading this page in the future and you want to check whether 14971 is still the latest build for PC:

https://blogs.windows.com/blog/tag/windows-insider-program/#DjqYqOlwb3VU05QG.97

**Comment by tyoc213** [comment]
this thing still doesnt update :'( from my build... so I will keep waiting

**Comment by tyoc213** [comment]
Well it is now updated to the latest, and also I uninstalled the 14.04 LTS and installed the xenial one which causes https//rustup.rs to fail (I will report that later in other issue here).

And updated my comment on rust at tensorflow project https://github.com/tensorflow/rust/issues/36#issuecomment-267707612

I add the strace which I didnt put on my first report

[strace.txt](https://github.com/Microsoft/BashOnWindows/files/658411/strace.txt)


Also @aseering I did found https://buildfeed.net/ to be up to date with the numbers.

So, this will be reported as a bug? (I mean labeled?) or do I need to add more specific info?
